daemon/api_test.go

```diff
@@ -799,6 +801,11 @@ func (s *apiSuite) genericTestPackagePut(c *check.C, body io.Reader, concreteNo
 	c.Assert(task, check.NotNil)
 	c.Check(task.State(), check.Equals, TaskRunning)
 
+	// one for each installed package above
```

Comment:
It's not entirely clear what is being synchronized on, which means we'd benefit from at least a comment here explaining what this is all about. Is the goal here to ensure `task.State()` remains `Running` so the Check above works, rather than going straight into the `!= Running` state waited for below? If that's the case, perhaps we can just drop the check above altogether, since it's probably not related to the functionality under review in these tests, and thus avoid the sync.

If that's not what this is about, perhaps we can put sync.WaitGroup to good use, and avoid having the N count so far away from the logic motivating it. Dropping a line there will mean a silent deadlock due to a sender-without-receiver situation.
```
var wg sync.WaitGroup

wg.Add(n)
// ... n x mkInstalled ...
// ... stuff ...
wg.Wait()